drivers/flash/flash_sam0.c

```diff
@@ -146,8 +165,12 @@ static int flash_sam0_write_page(struct device *dev, off_t offset,
 static int flash_sam0_erase_row(struct device *dev, off_t offset)
 {
 	*FLASH_MEM(offset) = 0U;
+#ifdef NVMCTRL_CTRLA_CMD_ER
 	NVMCTRL->CTRLA.reg = NVMCTRL_CTRLA_CMD_ER | NVMCTRL_CTRLA_CMDEX_KEY;
-
+#else
+	/* TODO: is erase page the right thing to do here? */
+	NVMCTRL->CTRLB.reg = NVMCTRL_CTRLB_CMD_EP | NVMCTRL_CTRLB_CMDEX_KEY;
```

Comment:
I think you're looking for erase block (CMD_EB), "blocks" in Microchip speak here are the the minimum erasure unit for most of the flash ("page" in Microchip speak is the program unit, which looks like it's only available on the AUX flash (table 25-2)).

Additionally, from the line below that table, I think you need to adjust "PAGES_PER_ROW" to 16.  That is, I think SAMD5x just replaced "row" with "block" terminology wise.

Reply:
Again, thank you!